### ISSUE TITLE
Learning rate warmup can fail when doing continued training

### ISSUE THREAD

**Issue by tuglat** [awslabs/sockeye#925]
When performing continued training using the parameter '--learning-rate-warmup`, and the warmup value is greater than the number of updates updates that have been saved at the last checkpoint, the warmup count starts over again at 0. 

This is a result of the fact that the optimizer is not saved when saving training state. The optimizer contains internal state variables, 'begin_num_update' and 'num_update', that are reset to 0 when restarting w/ a new optimizer.
